dstore.go

```diff
@@ -21,118 +21,33 @@ type Queries interface {
 	GetSize() string
 }
 
+// Datastore is a SQL backed datastore.
 type Datastore struct {
 	db      *sql.DB
 	queries Queries
 }
 
-// NewDatastore returns a new datastore
+// NewDatastore returns a new SQL datastore.
 func NewDatastore(db *sql.DB, queries Queries) *Datastore {
 	return &Datastore{db: db, queries: queries}
 }
 
-type batch struct {
-	db      *sql.DB
-	queries Queries
-	txn     *sql.Tx
-}
-
-func (b *batch) GetTransaction() (*sql.Tx, error) {
-	if b.txn != nil {
-		return b.txn, nil
-	}
-
-	newTransaction, err := b.db.Begin()
-	if err != nil {
-		if newTransaction != nil {
-			// nothing we can do about this error.
-			_ = newTransaction.Rollback()
-		}
-
-		return nil, err
-	}
-
-	b.txn = newTransaction
-	return newTransaction, nil
-}
-
-func (b *batch) Put(key ds.Key, val []byte) error {
-	txn, err := b.GetTransaction()
-	if err != nil {
-		_ = b.txn.Rollback()
-		return err
-	}
-
-	_, err = txn.Exec(b.queries.Put(), key.String(), val)
-	if err != nil {
-		_ = b.txn.Rollback()
-		return err
-	}
-
-	return nil
-}
-
-func (b *batch) Delete(key ds.Key) error {
-	txn, err := b.GetTransaction()
-	if err != nil {
-		_ = b.txn.Rollback()
-		return err
-	}
-
-	_, err = txn.Exec(b.queries.Delete(), key.String())
-	if err != nil {
-		_ = b.txn.Rollback()
-		return err
-	}
-
-	return err
-}
-
-func (b *batch) Commit() error {
-	if b.txn == nil {
-		return errors.New("no transaction started, cannot commit")
-	}
-	var err = b.txn.Commit()
-	if err != nil {
-		_ = b.txn.Rollback()
-		return err
-	}
-
-	return nil
-}
-
-func (d *Datastore) Batch() (ds.Batch, error) {
-	batch := &batch{
-		db:      d.db,
-		queries: d.queries,
-		txn:     nil,
-	}
-
-	return batch, nil
-}
-
+// Close closes the underying SQL database.
 func (d *Datastore) Close() error {
 	return d.db.Close()
 }
 
+// Delete removes a row from the SQL database by the given key.
 func (d *Datastore) Delete(key ds.Key) error {
-	result, err := d.db.Exec(d.queries.Delete(), key.String())
+	_, err := d.db.Exec(d.queries.Delete(), key.String())
 	if err != nil {
 		return err
 	}
 
-	rows, err := result.RowsAffected()
-	if err != nil {
-		return err
-	}
-
-	if rows == 0 {
-		return ds.ErrNotFound
-	}
-
```

Comment:
Removed because leveldb, flatfs and [MemoryDatastore](https://github.com/ipfs/go-datastore/blob/master/basic_ds.go) all do not return an error.